pkg/applicationmanager/controllers/applicationbackup.go

```diff
@@ -2779,3 +2787,45 @@ func handleLargeResourceError(a *ApplicationBackupController, backup *stork_api.
 		}
 	}
 }
+
+func getBatchCountForGetBackupStatus() int {
+	if envValue := os.Getenv(getBackupStatusBatchCountEnvVar); len(envValue) != 0 {
+		batchCount, err := strconv.Atoi(envValue)
+		if err != nil {
+			return defaultBackupVolumeBatchCount
+		}
+		return batchCount
+	}
+	return defaultBackupVolumeBatchCount
+}
+
+// getAndUpdateBackupStatusInBatches - fetches the backup status of volumes in batches.
+// Additionally, it updates the last update timestamp of the backup CR for each batch to prevent backup timeouts in px-backup
+func (a *ApplicationBackupController) getAndUpdateBackupStatusInBatches(
+	backup *stork_api.ApplicationBackup,
+	driver volume.Driver,
+	driverName string,
+	batchCount int,
+) ([]*stork_api.ApplicationBackupVolumeInfo, error) {
+	volumes := backup.Status.Volumes
+	volumeInfos := make([]*stork_api.ApplicationBackupVolumeInfo, 0)
+	// Fetch the status of the volumes in batches
+	for i := 0; i < len(volumes); i += batchCount {
+		end := min(i+batchCount, len(volumes))
+		batchVolumes := volumes[i:end]
+		backup.Status.Volumes = batchVolumes
+		batchVolumeInfos, err := driver.GetBackupStatus(backup)
+		if err != nil {
+			log.ApplicationBackupLog(backup).Errorf("failed to get volume status for driver %v: %v", driverName, err)
+			return nil, err
+		}
+		backup.Status.LastUpdateTimestamp = metav1.Now()
+		backup.Status.Volumes = volumes
```

Comment:
Will it be an issue, if we assign backup.Status.Volumes = volumeInfos and update the CR? That way, we will be push the latest status of volume as well. Any we have volumes variable, which is old content on which for loop will work on.